tools/health-monitor/README.md

```diff
@@ -0,0 +1,99 @@
+# Blobfuse2 Health Monitor (Preview)
+
+## About
+
+Blobfuse2 Health Monitor is a tool which will help in monitoring the mounts done via Blobfuse2. It supports the following types of monitors:
+
+1. **Blobfuse2 Stats Monitor:** Monitor the different statistics of blobfuse2 components like,
+    - Total bytes uploaded and downloaded via blobfuse2
+    - Events like create, delete, rename, synchronize, truncate, etc. on files or directories in the mounted directory
+    - Progress of uploads or downloads of large files to/from Azure Storage
+    - Keep track of number of calls that were made to Azure Storage for operations like create, delete, rename, chmod, etc. in the mounted directory
+    - Total number of open handles on files
+    - Number of times an open file request was served from the file cache or downloaded from the Azure Storage  
+
+2. **CPU and Memory Monitor:** Monitor the CPU and memory usage of the Blobfuse2 process associated with the mount
+
+3. **File Cache Monitor:** Monitor the file cache directory specified while mounting. This monitor does the following,
+    - Monitor the different events like create, delete, rename, chmod, etc. of files and directories in the cache
+    - Keep track of the cache consumption with respect to the cache size specified during mounting
+
+> **Note:** Health Monitor runs as a separate process where one health monitor process is associated with monitoring one blobfuse2 mounted directory.
+
+## Enable Health Monitor
+
+The different configuration options for the health monitor are,
+- `enable-monitoring: true|false`: Boolean parameter to enable health monitor. By default it is disabled
+- `stats-poll-interval-sec: <TIME IN SECONDS>`: Blobfuse2 stats polling interval (in sec). Default is 10 seconds
+- `process-monitor-interval-sec: <TIME IN SECONDS>`: CPU and memory usage polling interval (in sec). Default is 30 sec
+- `output-path: <PATH>`: Path where health monitor will generate its output file. It takes the current directory as default, if not specified. Output file name will be `monitor_<pid>.json`
+- `monitor-disable-list: <LIST OF MONITORS>`: List of monitors to be disabled. To disable a monitor, add its corresponding name in the list
+    - `blobfuse_stats` - Disable blobfuse2 stats polling
+    - `cpu_profiler` - Disable CPU monitoring on blobfuse2 process
+    - `memory_profiler` - Disable memory monitoring on blobfuse2 process
+    - `cache_monitor` - Disable file cache directory monitor
```

Comment:
Should we explicitly rename this to file cache monitor? Cause in the future we could have a stream cache monitor or something right?

Reply:
we may not have stream cache monitor as such (as this is just monitoring a directory events and streaming will rather be part of the stats-monitoring), but still renaming this to file_cache_monitor makes more sense.

Reply:
Renamed to file_cache_monitor